### ISSUE TITLE
react/no-unused-prop-types false positive when sharing an interface between components

### ISSUE THREAD

**Issue by dartess** [yannickcr/eslint-plugin-react#3270]
(and is not fully used in all components)

```ts
import React from 'react';

type Props = {
  method: string;
  isMobile: boolean; // ESLint: 'isMobile' PropType is defined but prop is never used(react/no-unused-prop-types)
};

function Component(props: Props) {
  const { method, isMobile } = props;
  return isMobile ? <MobileComponent {...props} /> : <>d: {method}</>;
}

function MobileComponent({ method }: Props) {
  return <>m: {method}</>;
}
```

**Comment by ljharb** [comment]
The error seems correct to me. These components shouldn't share the same type, because they don't use the same props.

There should be two separate Props types (one composed of the other, perhaps).

**Comment by dartess** [comment]
It makes sense
